RecoParticleFlow/PFClusterProducer/plugins/PFRecHitProducer.cc

```diff
@@ -54,9 +55,18 @@ void
 
    out->reserve(localRA1.upper());
    cleaned->reserve(localRA2.upper());
+
+   if (!init_) { // should go in beginRun
+     for( const auto& creator : creators_ ) {
+       creator->init(iSetup);
```

Comment:
ok, added the method of beginLuminosityBlock in PFRecHitProducer.h and put the init function there